<launch>
    <!-- Task and logging arguments -->
    <arg name="task_type"/>
    <arg name="logging_enabled"                 default="true"/>
    <arg name="test_id"                         default="default"/>
    <arg name="screenshots_enabled"             default="false"/>

    <!-- Simulator arguments -->
    <arg name="feedback_covariance"             default="0.0"/>
    <arg name="start_bullet_viewer"             default="true"/>

    <!-- Planner model arguments -->
    <arg name="multi_model"                     default="false"/>
    <arg name="deformability_override"          default="false"/>
    <arg name="use_adaptive_model"              default="false"/>
    <arg name="adaptive_model_learning_rate"    default="0.000001"/>
    <arg name="calculate_regret"                default="false"/>

    <!-- Planner internal arguments -->
    <arg name="planning_horizon"                default="1"/>
    <arg name="optimization_enabled"            default="false"/>
    <arg name="kalman_parameters_override"      default="false"/>

    <param name="logging_enabled" value="$(arg logging_enabled)"/>
    <param name="log_folder" value="$(find smmap)/../logs/$(arg task_type)/$(arg test_id)/"/>

    <include file="$(find smmap)/launch/$(arg task_type)_params.launch"/>

    <include file="$(find deform_simulator)/launch/deform_simulator.launch">
        <arg name="feedback_covariance" value="$(arg feedback_covariance)"/>
        <arg name="start_bullet_viewer" value="$(arg start_bullet_viewer)"/>
        <arg name="screenshots_enabled" value="$(arg screenshots_enabled)"/>
    </include>

    <group ns="smmap_planner_node">
        <param name="planning_horizon"                  value="$(arg planning_horizon)"             type="int"/>
        <param name="optimization_enabled"              value="$(arg optimization_enabled)"         type="bool"/>
        <param name="use_multi_model"                   value="$(arg multi_model)"                  type="bool"/>
        <param name="translational_deformability"       value="$(arg translational_deformability)"  type="double"   if="$(arg deformability_override)"/>
        <param name="rotational_deformability"          value="$(arg rotational_deformability)"     type="double"   if="$(arg deformability_override)"/>
        <param name="use_adaptive_model"                value="$(arg use_adaptive_model)"           type="bool"/>
        <param name="adaptive_model_learning_rate"      value="$(arg adaptive_model_learning_rate)" type="double"/>
        <param name="process_noise_factor"              value="$(arg process_noise_factor)"         type="double"   if="$(arg kalman_parameters_override)"/>
        <param name="observation_noise_factor"          value="$(arg observation_noise_factor)"     type="double"   if="$(arg kalman_parameters_override)"/>
        <param name="calculate_regret"                  value="$(arg calculate_regret)"             type="bool"/>
    </group>
</launch>
